src/test/test-generic-netlink-manual.c

```diff
@@ -0,0 +1,130 @@
+        uint16_t type;
+        int r;
+
+        assert_se(sd_generic_netlink_resolve_family(rtnl, FOU_GENL_NAME, &type) >= 0);
+
+        assert_se(sd_rtnl_message_new_generic_netlink(type, FOU_CMD_ADD, FOU_GENL_VERSION, &m) >= 0);
+        assert_se(sd_generic_netlink_message_append_u16(m, FOU_ATTR_PORT, htobe16(port)) >= 0);
+        assert_se(sd_generic_netlink_message_append_u8(m, FOU_ATTR_TYPE, FOU_ENCAP_GUE) >= 0);
+        assert_se(sd_generic_netlink_message_append_u16(m, FOU_ATTR_AF, AF_INET) >= 0);
+        assert_se(sd_generic_netlink_message_append_u8(m, FOU_ATTR_IPPROTO, 4) >= 0);
+
+        r = sd_netlink_call(rtnl, m, 0, NULL);
+        if (r < 0)
+                printf("Failed to add FooOverUDP port: %d \t %s\n", port, strerror(-r));
+
+        assert_se(r >= 0);
+
+        m = sd_netlink_message_unref(m);
+
+        return EXIT_SUCCESS;
+}
+
+static int generic_netlink_family_get(sd_netlink *rtnl) {
+        struct rtattr *tb[CTRL_ATTR_MAX + 1] = {};
+        sd_netlink_message *m = NULL, *n = NULL;
+        char *family_name = NULL;
+        uint16_t family;
+        int r;
+
+        assert_se(sd_rtnl_message_new_generic_netlink(GENL_ID_CTRL, CTRL_CMD_GETFAMILY, 1, &m) >= 0);
+        assert_se(sd_generic_netlink_message_append_u16(m, CTRL_ATTR_FAMILY_ID, GENL_ID_CTRL) >= 0);
+        assert_se(sd_generic_netlink_message_append_string(m, CTRL_ATTR_FAMILY_NAME, FOU_GENL_NAME) >= 0);
+
+        assert_se(sd_netlink_call(rtnl, m, 0, &n) >= 0);
+
+        assert_se(sd_generic_netlink_message_parse(n, tb) >= 0);
+
+        if (tb[CTRL_ATTR_FAMILY_NAME]) {
+                r =  sd_generic_netlink_message_read_string(tb, CTRL_ATTR_FAMILY_NAME, &family_name);
+                if (r >= 0)
+                        printf("name=%s\t", family_name);
+        }
+
+        if (tb[CTRL_ATTR_FAMILY_ID]) {
+                r = sd_generic_netlink_message_read_u16(tb, CTRL_ATTR_FAMILY_ID, &family);
+                if (r >= 0)
+                        printf("family id=%u\t", family);
+        }
+
+        if (tb[CTRL_ATTR_OPS]) {
+                struct rtattr *tb_ops[CTRL_ATTR_OP_MAX +1] = {};
+                struct rtattr *rta;
+                uint32_t op_id;
+
+                rta = tb[CTRL_ATTR_OPS];
+
+                r = sd_generic_netlink_message_parse_container(rta, tb_ops);
+                assert_se(r >= 0);
+
+                if (r >= 0) {
+                        if (tb_ops[CTRL_ATTR_OP_ID]) {
+                                r = sd_generic_netlink_message_read_u32(tb, CTRL_ATTR_OP_ID, &op_id);
+                                assert_se(r >= 0);
+                                if (r >= 0)
+                                        printf("id-0x%x ", op_id);
+                        }
+                }
+
+                printf("\n");
+        }
+        printf("\n");
+
+        assert_se((m = sd_netlink_message_unref(m)) == NULL);
+        assert_se((n = sd_netlink_message_unref(n)) == NULL);
+
+        return EXIT_SUCCESS;
```

Comment:
`return 0`. We use EXIT_* only for values which are (eventually) returned by `main()`.